x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/schema_meta_engine_logic.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { SchemaBaseLogic, SchemaBaseValues, SchemaBaseActions } from './schema_base_logic';
+import { MetaEngineSchemaApiResponse } from './types';
+
+interface MetaEngineSchemaValues extends SchemaBaseValues {
+  fields: MetaEngineSchemaApiResponse['fields'];
+  conflictingFields: MetaEngineSchemaApiResponse['conflictingFields'];
+  conflictingFieldsCount: number;
+  hasConflicts: boolean;
+}
+
+interface MetaEngineSchemaActions extends SchemaBaseActions {
+  loadMetaEngineSchema(): void;
+  onMetaEngineSchemaLoad(response: MetaEngineSchemaApiResponse): MetaEngineSchemaApiResponse;
+}
+
+export const MetaEngineSchemaLogic = kea<
+  MakeLogicType<MetaEngineSchemaValues, MetaEngineSchemaActions>
+>({
+  path: ['enterprise_search', 'app_search', 'meta_engine_schema_logic'],
+  connect: {
+    values: [SchemaBaseLogic, ['dataLoading', 'schema']],
+    actions: [SchemaBaseLogic, ['loadSchema', 'setSchema']],
+  },
+  actions: {
+    loadMetaEngineSchema: true,
+    onMetaEngineSchemaLoad: (response) => response,
+  },
+  reducers: {
+    fields: [
+      {},
+      {
+        onMetaEngineSchemaLoad: (_, { fields }) => fields,
+      },
+    ],
+    conflictingFields: [
+      {},
+      {
+        onMetaEngineSchemaLoad: (_, { conflictingFields }) => conflictingFields,
+      },
+    ],
+  },
+  selectors: {
+    conflictingFieldsCount: [
+      (selectors) => [selectors.conflictingFields],
+      (conflictingFields) => Object.keys(conflictingFields).length,
+    ],
+    hasConflicts: [
+      (selectors) => [selectors.conflictingFieldsCount],
+      (conflictingFieldsCount) => conflictingFieldsCount > 0,
+    ],
```

Comment:
FYI that a lot of these selectors were in-component vars in the standalone UI but I thought it would be nice to make these handy dandy selectors. Maybe a lil overkill, but I like it 🤔

Reply:
I've been doing this a bit as well, I like it. I feel like it slims up the components and make them a bit more readable. It's also a bit more performant since selectors are all memoized.

Reply:
This is a great decision. So much better. Thanks for commenting here Constance